scipy/special/tests/test_support_alternative_backends.py

```diff
@@ -75,6 +69,17 @@ def test_support_alternative_backends(xp, data, f_name_n_args):
     args_np = [np.asarray(data.draw(npst.arrays(dtype_np, shape, elements=elements)))
                for shape in shapes]
 
+    # Make exceptions for counterexamples
+    if is_jax(xp):
+        if f_name in {'gammainc', 'gammaincc'}:  # google/jax#20507
+            pytest.skip("google/jax#20507")
```

Comment:
It seems there may be other edge cases worth reporting here. I can't be sure I have all the `assume`s we need without iterative testing, so skip them all for now. @lucascolley is this something you'd be willing to follow up on?

Reply:
changing this to `assume(np.all(a > 1e-30))` reduces to only one failure: `gammaincc(1, -1)`

```
ref        = Array(nan, dtype=float32)
res        = Array(1., dtype=float32)
```

Also, a `relentr` failure which seems flaky?

<details> 

```
_______________________________________________________________________ test_support_alternative_backends[f_name_n_args6-jax.numpy] ________________________________________________________________________
scipy/special/tests/test_support_alternative_backends.py:51: in test_support_alternative_backends
    @array_api_compatible
        f          = <function given.<locals>.run_test_as_given.<locals>.wrapped_test at 0x14545f060>
        f_name_n_args = ('rel_entr', 2)
        xp         = <module 'jax.numpy' from '/Users/lucascolley/mambaforge/envs/scipy-jax/lib/python3.12/site-packages/jax/numpy/__init__.py'>
scipy/special/tests/test_support_alternative_backends.py:98: in test_support_alternative_backends
    xp_assert_close(res, ref, rtol=eps**0.2, atol=eps*20,
        args_np    = [array(0., dtype=float32), array(0.5, dtype=float32)]
        args_xp    = [Array(0., dtype=float32), Array(0.5, dtype=float32)]
        data       = data(...)
        dtype      = 'float32'
        dtype_np   = <class 'numpy.float32'>
        dtype_xp   = <class 'jax.numpy.float32'>
        elements   = {'allow_subnormal': False, 'max_value': 10.0, 'min_value': -10.0}
        eps        = 1.1920929e-07
        f          = <function support_alternative_backends.<locals>.wrapped at 0x1169d0720>
        f_name     = 'rel_entr'
        f_name_n_args = ('rel_entr', 2)
        final_shape = ()
        mbs        = mutually_broadcastable_shapes(num_shapes=2)
        n_args     = 2
        ref        = Array(0., dtype=float32)
        res        = Array(0.5, dtype=float32)
        shapes     = ((), ())
        x          = array(0., dtype=float32)
        xp         = <module 'jax.numpy' from '/Users/lucascolley/mambaforge/envs/scipy-jax/lib/python3.12/site-packages/jax/numpy/__init__.py'>
        y          = array(0.5, dtype=float32)
../../../../../../mambaforge/envs/scipy-jax/lib/python3.12/contextlib.py:81: in inner
    return func(*args, **kwds)
E   AssertionError: 
E   Not equal to tolerance rtol=0.0412346, atol=2.38419e-06
E   
E   Mismatched elements: 1 / 1 (100%)
E   Max absolute difference: 0.5
E   Max relative difference: inf
E    x: array(0.5, dtype=float32)
E    y: array(0., dtype=float32)
E   Falsifying example: test_support_alternative_backends(
E       xp=<module 'jax.numpy' from '/Users/lucascolley/mambaforge/envs/scipy-jax/lib/python3.12/site-packages/jax/numpy/__init__.py'>,
E       f_name_n_args=('rel_entr', 2),
E       data=data(...),
E   )
E   Draw 1: BroadcastableShapes(input_shapes=((), ()), result_shape=())
E   Draw 2: 'float32'
E   Draw 3: array(0., dtype=float32)
E   Draw 4: array(0.5, dtype=float32)
E   Explanation:
E       These lines were always and only run by failing examples:
E           /Users/lucascolley/mambaforge/envs/scipy-jax/lib/python3.12/site-packages/numpy/core/arrayprint.py:1012
E           /Users/lucascolley/mambaforge/envs/scipy-jax/lib/python3.12/site-packages/numpy/core/arrayprint.py:1013
E           /Users/lucascolley/mambaforge/envs/scipy-jax/lib/python3.12/site-packages/numpy/core/arrayprint.py:1020
E           /Users/lucascolley/mambaforge/envs/scipy-jax/lib/python3.12/site-packages/numpy/core/arrayprint.py:1021
E           /Users/lucascolley/mambaforge/envs/scipy-jax/lib/python3.12/site-packages/numpy/core/arrayprint.py:108
E           (and 11 more with settings.verbosity >= verbose)
E   
E   You can reproduce this example by temporarily adding @reproduce_failure('6.103.2', b'AXicY2AAAUYGZADmHYDxAAcMAMM=') as a decorator on your test case

```

</details>

Reply:
> but it was there before.

We may want to do something to prevent one failure from changing the other tests that get run.

https://hypothesis.readthedocs.io/en/latest/details.html#making-random-code-deterministic

I suppose this is not worth a ton of time, though. The purpose of these tests is just to make sure that dispatch is going to the correct function with the right argument order, not to look for counterexamples. It works now, and it would be difficult for someone to make a change to the code such that dispatch to JAX `gammainc`/`gammaincc` is the ONLY thing that breaks, so maybe safe enough to leave this as is. Maybe update your issues to JAX with any findings, though.

Reply:
`gammaincc(1, -1) == 1` looks like another bug in JAX - the formula in the documentation is defined for non-negative `x` only.

Reply:
Okay cool, that's also fixed in Pearu's PR for the other `gamma` problem.

Reply:
I think whenever we see a failure in CuPy with something on the order of `1e-38`, it's because CuPy doesn't seem to support `float32` subnormal numbers:
```python3
import  numpy as np
import cupy as cp

x = np.finfo(np.float64).smallest_normal  # 1.1754944e-38
(np.asarray([x]) / 2)  # array([5.877472e-39], dtype=float32)
(cp.asarray([x]) / 2)  # array([0.], dtype=float32)
```
Weird.